app/upgrade_cudos.go

```diff
@@ -881,39 +879,30 @@ func createDelegation(ctx sdk.Context, app *App, originalValidator string, newDe
 }
 
 func fundCommunityPool(ctx sdk.Context, app *App, genesisData *GenesisData, cudosCfg *CudosMergeConfig, manifest *UpgradeManifest) error {
-	// Fund community pool
+	// Get addresses and amounts
+	cudosCommunityPoolFundsAccount := genesisData.accounts.MustGet(cudosCfg.config.RemainingDistributionBalanceAddr)
 	communityPoolBalance, _ := genesisData.distributionInfo.feePool.communityPool.TruncateDecimal()
 	convertedCommunityPoolBalance, err := convertBalance(communityPoolBalance, cudosCfg)
 	if err != nil {
 		return err
 	}
 
-	communityPoolSourceAccountRawAddress := genesisData.accounts.MustGet(cudosCfg.config.RemainingDistributionBalanceAddr).rawAddress
-
-	if cudosCfg.config.CommunityPoolBalanceDestAddr == "" {
-		// Move balance to community pool if destination address is not set
-
-		err = app.DistrKeeper.FundCommunityPool(ctx, convertedCommunityPoolBalance, communityPoolSourceAccountRawAddress)
-		if err != nil {
-			return err
-		}
-
-		registerBalanceMovement(cudosCfg.config.RemainingDistributionBalanceAddr, distrtypes.ModuleName, communityPoolBalance, convertedCommunityPoolBalance, "community_pool_balance", manifest)
-
-	} else {
-		// Move balance to given account
-
-		destAccRawAddr, err := convertAddressToRaw(cudosCfg.config.CommunityPoolBalanceDestAddr, cudosCfg)
-		if err != nil {
-			return err
-		}
+	// Mint balance to distribution leftover address
+	err = migrateToAccount(ctx, app, minttypes.ModuleName, cudosCommunityPoolFundsAccount.rawAddress, communityPoolBalance, convertedCommunityPoolBalance, "community_pool_balance", manifest)
+	if err != nil {
+		return err
+	}
 
-		err = app.BankKeeper.SendCoins(ctx, communityPoolSourceAccountRawAddress, destAccRawAddr, convertedCommunityPoolBalance)
-		if err != nil {
-			return err
-		}
-		registerBalanceMovement(cudosCfg.config.RemainingDistributionBalanceAddr, cudosCfg.config.CommunityPoolBalanceDestAddr, communityPoolBalance, convertedCommunityPoolBalance, "community_pool_balance", manifest)
+	// Move balance to destination chain community pool
+	err = app.DistrKeeper.FundCommunityPool(ctx, convertedCommunityPoolBalance, cudosCommunityPoolFundsAccount.rawAddress)
+	if err != nil {
+		return err
+	}
```

Comment:
Pls relocate also the other code flow case in this function (for the case when there is configured landing account for community pool balance).